src/rpcmessage.rs

```diff
@@ -74,37 +75,42 @@ impl RpcMessage {
         RpcFrame::from_rpcmessage(self)
     }
     pub fn param(&self) -> Option<&RpcValue> {
-        self.key(Key::Params as i32)
+        self.ival(Key::Params as i32)
     }
     pub fn set_param(&mut self, rv: impl Into<RpcValue>) -> &mut Self {
         self.set_param_opt(Some(rv.into()))
     }
     pub fn set_param_opt(&mut self, rv: Option<RpcValue>) -> &mut Self {
-        self.set_key(Key::Params, rv)
+        self.set_ival(Key::Params, rv)
     }
     pub fn result(&self) -> Result<&RpcValue, RpcError> {
-        match self.key(Key::Result as i32) {
+        match self.ival(Key::Error as i32) {
             None => {
-                match self.error() {
-                    None => {Err(RpcError{ code: RpcErrorCode::InternalError, message: "Neither 'result' nor 'error' key found in RPC response.".to_string() })}
-                    Some(err) => {Err(err)}
+                match self.ival(Key::Result as i32) {
+                    None => {
+                        Ok(STATIC_NULL_RPCVALUE.get_or_init(RpcValue::null))
+                    }
+                    Some(rv) => { Ok(rv) }
                 }
             }
-            Some(rv) => {Ok(rv)}
+            Some(err) => {
+                Err(RpcError::from_rpcvalue(err)
+                    .unwrap_or(RpcError{ code: RpcErrorCode::InvalidParam, message: "Cannot parse 'error' key found in RPC response.".to_string() }))
```

Comment:
You could refactor the RpcError in the unwrap_or to a const variable, and then, if you implement the other suggestion, you could make both the match arms oneline again